drivers/stepper/gpio_stepper/gpio_stepper_controller.c

```diff
@@ -0,0 +1,201 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2024 Carl Zeiss Meditec AG
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT gpio_stepper
+
+#include "zephyr/drivers/gpio.h"
+#include <zephyr/kernel.h>
+#include "zephyr/drivers/stepper.h"
+#include <zephyr/logging/log.h>
+#include "stdlib.h"
+
+LOG_MODULE_REGISTER(gpio_stepper_motor_controller, CONFIG_STEPPER_LOG_LEVEL);
+
+struct gpio_stepper_motor_controller_data {
+	struct k_work_delayable stepper_dwork;
+	const struct gpio_dt_spec *gpio_in_pins;
+	int32_t min_position;
+	int32_t max_position;
+	int32_t actual_position;
+	int32_t target_position;
+	uint32_t actual_speed;
+	uint32_t delay_in_us;
+	bool is_moving;
+};
+
+struct gpio_stepper_motor_controller_config {
+	const uint16_t motor_steps_per_rev;
+	const double motor_gear_ratio;
+	const uint8_t number_in_pins;
```

Comment:
This value seems redundant as it has to be 4 for the driver to work, just have a build assert which ensures the gpio array is 4 and remove the checks from the driver :)

Reply:
Hmm this is actually a placeholder, coz there are gpio motors which have to driven with 5 gpios rather than 4. But I'll remove it for now